mx.py

```diff
@@ -3590,6 +3590,8 @@ def update(self, vcdir, rev=None, mayPull=False, clean=False, abortOnError=False
         """
         if rev and mayPull and not self.exists(vcdir, rev):
             self.pull(vcdir, rev=rev, update=False, abortOnError=abortOnError)
+            if not self.exists(vcdir, rev):
+                abort('Fetch of %s succeeded\nbut did not contain requested revision %s.\nCheck that your remotes agree with the suite.py reposity location' % (vcdir, rev))
```

Comment:
Fixed the typo.  The only way to get into the state where your origin and the suite.py location are out of agreement is if you know something about using git.  So I think telling them to check their remotes using git remote -v should be sufficient, so I'l update the text a little bit.  The text will now read 
<pre>
$ mx sfor
Fetch of /Users/tkrodrig/ws/basic-graal/truffle succeeded
but did not contain requested revision e322868c76f775a399bf96915c95e077c4f2c08c.
Check that the suite.py repository location is mentioned by 'git remotes -v
</pre>

I don't think that we should do a direct fetch for them either.  git really wants you to do this through your configured remotes so I think we should treat this as a misconfiguration.  We also don't know the proper URL down inside the VC.